subsys/bluetooth/host/hci_core.c

```diff
@@ -3780,10 +3787,29 @@ int bt_disable(void)
 
 	bt_monitor_send(BT_MONITOR_CLOSE_INDEX, NULL, 0);
 
+#if defined(CONFIG_BT_EXT_ADV) || defined(CONFIG_BT_BROADCASTER)
+	bt_adv_reset_adv_pool();
+#endif /* CONFIG_BT_EXT_ADV || CONFIG_BT_BROADCASTER */
+
+#if defined(CONFIG_BT_PRIVACY)
+	k_work_cancel_delayable(&bt_dev.rpa_update);
+#endif /* CONFIG_BT_PRIVACY */
+
 #if defined(CONFIG_BT_PER_ADV_SYNC)
 	bt_periodic_sync_disable();
 #endif /* CONFIG_BT_PER_ADV_SYNC */
 
+#if defined(CONFIG_BT_CONN)
+#if defined(CONFIG_BT_SMP)
+	bt_pub_key_hci_disrupted();
+#endif /* CONFIG_BT_SMP */
```

Comment:
Please use `if (IS_ENABLED(CONFIG_...) {` whenever you can. In this case it should be possible